xtable-core/src/main/java/org/apache/xtable/delta/DeltaConversionTarget.java

```diff
@@ -258,10 +258,15 @@ public Optional<String> getTargetCommitIdentifier(String sourceIdentifier) {
         Optional<TableSyncMetadata> optionalMetadata =
             TableSyncMetadata.fromJson(sourceMetadataJson.get());
         if (!optionalMetadata.isPresent()) {
-          return Optional.empty();
+          continue;
         }
 
         TableSyncMetadata metadata = optionalMetadata.get();
+        // For backward compatability since the older version JSON might not contain new fields
```

Comment:
I think this will be covered at line 270 so we don't need the extra condition right?

Is there a test case where we will have a `null` to cover this case?